tangelo/toolboxes/qubit_mappings/combinatorial.py

```diff
@@ -0,0 +1,234 @@
+    in the Fock space.
+
+    Args:
+        ferm_op (FermionOperator). Fermionic operator, with alternate ordering
+            as followed in the openfermion package
+        n_modes (int): Number of relevant molecular orbitals, i.e. active molecular
+            orbitals.
+        n_electrons (int): Number of active electrons.
+
+    Returns:
+        QubitOperator: Self-explanatory.
+    """
+
+    # The chemist ordering seperates some 1-body and 2-body terms.
+    ferm_op_chemist = chemist_ordered(ferm_op)
+
+    # Specify the number of alpha and beta electrons.
+    if isinstance(n_electrons, tuple) and len(n_electrons) == 2:
+        n_alpha, n_beta = n_electrons
+    elif isinstance(n_electrons, int) and n_electrons % 2 == 0:
+        n_alpha = n_beta = n_electrons // 2
+    else:
+        raise ValueError(f"{n_electrons} is not a valid entry for n_electrons, must be a tuple or an int.")
+
+    # Get the number of qubits n.
+    n_choose_alpha = comb(n_modes, n_alpha, exact=True)
+    n = ceil(np.log2(n_choose_alpha * comb(n_modes, n_beta, exact=True)))
+
+    # Construct the basis set where each configutation is mapped to a unique
+    # integer.
+    basis_set_alpha = basis(n_modes, n_alpha)
+    basis_set_beta = basis(n_modes, n_beta)
+    basis_set = OrderedDict()
+    for sigma_alpha, int_alpha in basis_set_alpha.items():
+        for sigma_beta, int_beta in basis_set_beta.items():
+            # Alternate ordering (like FermionOperator in openfermion).
+            sigma = tuple(sorted([2*sa for sa in sigma_alpha] + [2*sb+1 for sb in sigma_beta]))
+            unique_int = (int_alpha * n_choose_alpha) + int_beta
+            basis_set[sigma] = unique_int
+
+    qu_op = QubitOperator()
+    # Check what is the effect of every term.
+    for term, coeff in ferm_op_chemist.terms.items():
+        # Core term.
+        if not term:
+            qu_op += coeff
+            continue
+
+        # Get the effect of each operator to the basis set items.
+        for b, unique_int in basis_set.items():
+
+            new_state, phase = one_body_op_on_state(term[-2:], b)
+
+            if len(term) == 4 and new_state:
+                new_state, phase_two = one_body_op_on_state(term[:2], new_state)
+                phase *= phase_two
+
+            if not new_state:
+                continue
+
+            new_unique_int = basis_set[new_state]
+            qu_op += element_to_qubitop(n, unique_int, new_unique_int, phase*coeff)
+
+    return qu_op
+
+
+def element_to_qubitop(n_qubits, i, j, coeff=1.):
+    """Map a matrix element to a qubit operator.
+
+    Args:
+        n_qubits (int): Self-explanatory.
+        i (int): i row of the matrix element.
+        j (int): j column of the matrix element.
+        coeff (complex): Value at position i,j in the matrix.
+
+    Returns:
+        QubitOperator: Self-explanatory.
+    """
+
+    # Must add 2 to the padding because of the "0b" prefix.
+    bin_i = format(i, f"#0{n_qubits+2}b")
```

Comment:
Is the goal to get the binary string representing `i` ?
If yes, why not just `bin(i)[2:]` ?

Reply:
Yes, but we need the padding as we want a constant-size string (`n_qubits` elements).